browser/brave_shields/reduce_language_navigation_throttle.cc

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_shields/reduce_language_navigation_throttle.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/feature_list.h"
+#include "base/threading/thread_task_runner_handle.h"
+#include "brave/browser/brave_browser_process.h"
+#include "brave/components/brave_shields/browser/brave_farbling_service.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "brave/components/brave_shields/common/features.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/language/core/browser/language_prefs.h"
+#include "components/language/core/browser/pref_names.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "net/http/http_util.h"
+#include "third_party/blink/renderer/platform/language.h"
+#include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
+
+namespace brave_shields {
+
+// static
+std::unique_ptr<ReduceLanguageNavigationThrottle>
+ReduceLanguageNavigationThrottle::MaybeCreateThrottleFor(
+    content::NavigationHandle* navigation_handle,
+    HostContentSettingsMap* content_settings) {
+  if (!base::FeatureList::IsEnabled(
```

Comment:
shouldn't this also be checking the pref? I know it's covered in ShouldDoReduceLanguage below, but I think we should combine the pref and feature into IsReduceLanguageEnabled and use that here, BraveShieldsWebContentsObserver and in ShouldDoReduceLanguage so they are all consistent

Reply:
changed in https://github.com/brave/brave-core/pull/12234/commits/740aa952789f965e57fb88a94a9efc31537d6b70

Reply:
Only reference to `features::kBraveReduceLanguage` (outside its definition) is in `components/brave_shields/browser/brave_shields_util.cc` new function `IsReduceLanguageEnabledForProfile` which is called from everywhere.